### PR TITLE
Store active users in state

### PR DESCRIPTION
**Pull request by SawyerHood**
This stores the active users in state. Right now this widget has non-determinism since active users can change while the widget is running. Now we snapshot the active users on insert and only update if you press the update users button.

### CODE REVIEW

File: code.tsx [hunk 1]
```diff
@@ -194,12 +195,22 @@ function getFilteredUsers(users: Array<User>) {
 }
 
 function Widget() {
-  const users = figma.activeUsers;
+  const [users, setUsers] = useSyncedState<ActiveUser[]>("users", []);
```
Comment (isg) [inline code.tsx:198]:
```suggestion
  const [users, setUsers] = useSyncedState<ActiveUser[]>("users", () => { updateUser() });
```

File: code.tsx [hunk 2]
```diff
@@ -248,6 +259,12 @@ function Widget() {
     }
   );
 
+  useEffect(() => {
+    if (users.length === 0) {
+      updateUsers();
+    }
+  });
```
Comment (isg) [inline code.tsx:266]: why do you need `useEffect`? Instead let's initialize it in `useSyncedState`?

Comment (SawyerHood) [reply code.tsx:266]: You are not allowed to do that atm

**Comment by elynnlee** [comment]
Thank you!! 🙏 🙏 🙏

**Pull request merged**